libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SalesforceNetworkPlugin.java

```diff
@@ -125,7 +125,9 @@ public void onSuccess(RestRequest request, RestResponse response) {
                     try {
                         // Not a 2xx status
                         if (!response.isSuccess()) {
-                            callbackContext.error(response.asString());
+                            JSONObject errorObj = new JSONObject();
+                            errorObj.putOpt("response", response.fullResponseAsJSONObject());
```

Comment:
Use `response.asJsonObject()` instead. Also, use `put()` instead of `putOpt()`, `null` as a value is fine.